architecture/crds.md

```diff
@@ -0,0 +1,324 @@
+# CRD Handling in Helm
+
+This document talks about the problems the Helm team has had dealing with CRDs, and lays out criterion for how to move forward. While it discusses a few solution _paths_, it does not provide a single solution. It is also an instrument for ruling out solutions that do not match Helm's guiding principles.
+
+The most intractable problem in Helm's history has been how to handle Kubernetes CRDs. We've tried a variety of approaches, none of which has proven satisfactory to all users. Our current solution, while not comprehensive, is designed to privilege safety over flexibility. We are considering options for a Helm 4 time frame, and this document is a first step in that exercise. (Backward-compatible features for CRDs could still be merged into Helm 3.)
```

Comment:
I suggest _An intractable_ or better yet _A difficult_ rather than _The most intractable_

Reply:
_all_ leans toward the hyperbolic. Perhaps rephrasing as _We have yet to find an approach that satisfies enough of the Helm user base._